### ISSUE TITLE
cmd/go: confusing error message when flags come before package paths

### ISSUE THREAD

**Issue by josharian** [golang/go#56310]
```txtar
-- go.mod --
module m

go 1.19
-- x/t_test.go --
package x

// some test code that has a -foo flag
```

(Warning: hand-written, untested txtar.)

```
$ go test ./x -foo
<OK>
$ go test -foo ./x
no Go files in /current/pwd
```

This is a confusing error message. I think(?) the Go tool has interpreted `-foo` as a package path, but the error message doesn't reflect that. It's true that there are no Go files in the root, but that's not the problem.

Maybe the Go command could make these two equivalent. But even if it can't, a more useful error message would be welcome.

**Comment by bcmills** [comment]
> I think(?) the Go tool has interpreted `-foo` as a package path, but the error message doesn't reflect that.

Nope: it has interpreted `-foo ./x` as the flag `-foo` with argument `./x`, leaving an implicit package path of “`.`”.

I agree that the diagnostic could perhaps be improved, though.

**Comment by josharian** [comment]
Oof.

Why should cmd/go assume that -foo takes an arg, as opposed to being a bool flag?

**Comment by bcmills** [comment]
It doesn't know, so it has to make a conservative guess. (FWIW, the situation was even worse before https://go.dev/cl/211358 and https://go.dev/cl/248618. 😅)
